### PR TITLE
Uses dashes to separate words in URLs

### PR DESCRIPTION
**Pull request by paulrobertlloyd**
The [URL standards for GOV.UK](https://www.gov.uk/guidance/content-design/url-standards-for-gov-uk) state that:

> dashes should be used to separate words within URLs so they are easy to read - for example, <https://www.gov.uk/set-up-business> (this might not apply if the URL is designed to be read aloud)

We should follow the same URL standards as used on GOV.UK and many other government services.

### CODE REVIEW

File: app/models/form.rb [hunk 1]
```diff
@@ -12,11 +12,11 @@ def initialize(form_path)
   end
 
   def get_subsection(id)
-    subsections.find { |s| s.id == id }
+    subsections.find { |s| s.id == id.to_s.underscore }
```
Comment (baarkerlounger) [inline app/models/form.rb:15]: Might be better to just expect them to be dasherized in the form definition JSON? We do control that spec after all.

**Review by baarkerlounger — APPROVED** [review]

File: app/models/form.rb (hunk 1 above)
Comment (paulrobertlloyd) [reply app/models/form.rb:15]: I tried that approach initially, but that involved changing more areas of the codebase (i.e. things like `question.hint_text` would need to be `question["hint-text"]`). But am happy with either approach.

Comment (baarkerlounger) [reply app/models/form.rb:15]: Wouldn't have to be consistent for all keys in the JSON, just page ids (although I appreciate that makes a find and replace fiddly). Not particularly important I think this is fine to go in as is.

Comment (paulrobertlloyd) [reply app/models/form.rb:15]: Underscores versus dashes is something I go back and forwards on… pros and cons to each approach, as is so often the case 🤪

**Pull request merged**